btb/hyper_parameter.py

```diff
@@ -103,25 +133,23 @@ def inverse_transform(self, x):
 
 
 class IntExpHyperParameter(FloatExpHyperParameter):
-    def __init__(self, typ, rang):
-        # can't use super() because we need to provide a cast explicitly
-        HyperParameter.__init__(self, rang, lambda x: math.log10(int(x)))
-
-    @property
-    def is_integer(self):
-        return True
+    param_type = ParamTypes.INT_EXP
+    is_integer = True
 
     def inverse_transform(self, x):
         return super(IntExpHyperParameter, self).inverse_transform(x).astype(int)
 
 
 class CatHyperParameter(HyperParameter):
-    def __init__(self, rang, cast):
-        self.cat_transform = {cast(each): 0 for each in rang}
+
+    def __init__(self, param_type=None, rang=None):
+        self.cat_transform = {self.cast(each): 0 for each in rang}
+
         # this is a dummy range until the transformer is fit
-        super(CatHyperParameter, self).__init__([0.0, 1.0], float)
+        super(CatHyperParameter, self).__init__(param_type, [0.0, 1.0])
 
     def fit_transform(self, x, y):
+        self.cat_transform_r = {each: (0, 0) for each in self.range}
```

Comment:
can we rename this attribute with a full name? is it `cat_transform_range`?

if conforming to sklearn-style estimator interface, attributes set in `fit` method should have a trailing `_`, like `self.cat_transform_range_` (http://scikit-learn.org/stable/developers/contributing.html#parameters-and-init)

Reply:
This actually was a development test that was left behind! I will clean it up.

For the trailing underscore thing, I think that we can tackle that in a separated PR, focused only on the scikit-learn API style.